xscen/indicators.py

```diff
@@ -193,7 +200,7 @@ def _infer_freq_from_meta(ind):
             if (out[c].attrs != ds[c].attrs) and (out[c].sizes == ds[c].sizes):
                 out[c].attrs = ds[c].attrs
 
-        if "time" in out.dims:
+        if restrict_years and "time" in out.dims:
```

Comment:
Could you add a test? What happens with something like `AS-JUL` ?

Reply:
I added a test with as-jul to show.
Basically, with xclim, you have a nan in the first and the last position.
The first time is the year before the start of the input.
The last time is the same year as the end of the input.
With `restrict_years`= False, we keep the time steps for the 2 nans.
With `restrict_years`= True, we cut the first nan.